### DISCUSSION TITLE
Variables in pipeline

### DISCUSSION THREAD

**Discussion by connormcmk** [nushell/nushell#6550] [Ideas]
Very frequently I find myself wanting to write something like: 
```
open clips.tsv | let clips
```

But it doesn't work, I get this help:
```
  help: Assigning 'value' to 'clips' does not produce a value to be piped. If the pipeline result is
        meant to be assigned to 'clips', use 'let clips = (value | ...)'.
```

But I don't want to write 
```
let clips = (value | ...) 
```

It's ugly, and it requires me to return to the beginning of the line to place parens, rather than continuously building up my pipeline. 

What I'd like is something like `name` which would let me give a value a variable name in a pipeline.

That would allow me to reverse the column order with something like this:
```
open clips.tsv | name clips | columns | reverse | name cols | $clips select $cols
```

**Comment by fdncred** [comment]
setting variables from a pipeline has been asked for before. I think it looks interesting to me and more nu-ish, imo.
